packages/ui/src/primitives/tailwind/Checkbox/index.tsx

```diff
@@ -23,60 +23,112 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
+const Checkbox = (
+  { checked, disabled, defaultChecked, indeterminate, label, description, onChange }: CheckboxProps,
+  ref: React.ForwardedRef<HTMLDivElement>
+) => {
   const handleChange = () => {
-    if (!disabled) {
-      onChange(!value)
+    if (!disabled && onChange) {
+      onChange(!checked)
     }
   }
 
-  const id = uuidv4()
+  const [isCheckedInternal, setIsCheckedInternal] = useState(defaultChecked !== undefined ? defaultChecked : checked)
+  const [isIndeterminateInternal, setIsIndeterminateInternal] = useState(indeterminate)
+
+  useEffect(() => {
+    if (checked) {
+      setIsCheckedInternal(true)
+      setIsIndeterminateInternal(false)
+    } else {
+      setIsCheckedInternal(false)
+      setIsIndeterminateInternal(indeterminate)
+    }
+  }, [checked])
+
+  useEffect(() => {
+    if (indeterminate) {
+      setIsCheckedInternal(false)
+      setIsIndeterminateInternal(true)
+    } else {
+      setIsIndeterminateInternal(false)
+      setIsCheckedInternal(checked)
+    }
+  }, [indeterminate])
 
   return (
-    <div className={twMerge('relative flex cursor-pointer items-end', containerClassName)}>
-      <input
-        type="checkbox"
-        checked={value}
-        onChange={handleChange}
-        id={id}
+    <div
+      className={twMerge(
+        'relative flex cursor-pointer items-center justify-center gap-x-2',
+        description && 'items-start'
+      )}
+    >
+      <div
+        className={twMerge(
+          'relative',
+          'grid h-4 w-4 place-items-center rounded',
+          'border border-[#42454D] bg-[#141619]',
+          !isCheckedInternal && !isIndeterminateInternal && !disabled && 'hover:border-[#9CA0AA] hover:bg-[#191B1F]',
+          !isCheckedInternal && !disabled && 'focus:border-[#375DAF] focus:bg-[#212226]',
+          (isCheckedInternal || isIndeterminateInternal) && 'border-[#375DAF] bg-[#212226]',
+          disabled && 'cursor-not-allowed border-[#42454D] bg-[#191B1F]'
+        )}
+        onClick={handleChange}
         onKeyDown={(e) => {
           if (e.key === 'Enter') {
             handleChange()
           }
         }}
-        className={twMerge(
-          'peer relative appearance-none',
-          'grid h-4 w-4 place-items-center rounded border border-theme-primary focus:border-2 focus:border-theme-focus focus:outline-none',
-          value ? 'bg-blue-primary' : 'bg-theme-surfaceInput',
-          disabled ? 'cursor-not-allowed opacity-50' : '',
-          className
-        )}
-      />
-      <HiCheck onClick={handleChange} className="absolute m-0.5 hidden h-3 w-3 text-white peer-checked:block" />
+        tabIndex={0}
+        ref={ref}
+      >
+        <FaCheck
+          onClick={handleChange}
+          className={twMerge(
+            'absolute h-3 w-3 transition-transform duration-200 ease-in-out',
+            disabled ? 'cursor-not-allowed text-[#42454D]' : 'text-[#5F7DBF]',
+            isCheckedInternal ? 'scale-100' : 'scale-0'
```

Comment:
we can use `checked && !indeterminate` instead of introducing another state

Reply:
internal state is used for following purpose:
- when indeterminate state is set to true: checked should be false (even if props.checked is set to true)
- when checked is set to true: indeterminate should be false (even if props.indeterminate is set to true)

so, newly set property takes precedence and overrides the old property. this logic is managed by internal states.